### PR TITLE
api,owner(ticdc): return error when api fails

### LINKED ISSUE #1710: changefeed remove has not any output even the operation failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Remove a changefeed that is not exist:
-
image

2. What did you expect to see?
stderr with msg, like "changefeed "R" is not exist, fail to remove. "
3. What did you see instead?

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       5.0.0-nightly
        ```

    - TiCDC version (execute `cdc version`):

        ```
        5.0.0-nightly
        ```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #1710 ref #3456

### What is changed and how it works?

API waits and returns owner response.

### Check List 

Tests 

 - Unit test

Side effects

 - Breaking backward compatibility (only for CLI)

### Release note 

```release-note
Synchronize HTTP API handling
```

### CODE REVIEW

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -201,58 +216,65 @@ func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState)
...
 	})
-	// wait the debug info printed
-	select {
-	case <-done:
-	case <-time.After(timeout):
-		fmt.Fprintf(w, "failed to print debug info for owner\n")
-	}
+}
+
+// Query writes debug info into the specified http writer
```
Comment (asddongmen) [inline cdc/owner/owner.go:263]: typos.

File: cdc/processor/manager.go [hunk 2]
```diff
@@ -141,31 +141,40 @@ func (m *Manager) closeProcessor(changefeedID model.ChangeFeedID) {
 
 // AsyncClose sends a close signal to Manager and closing all processors
```
Comment (asddongmen) [inline cdc/processor/manager.go:142]: // AsyncClose sends a signal to Manager to close all processors

File: cdc/api/open.go [hunk 3]
```diff
@@ -319,11 +319,16 @@ func (h *openAPI) PauseChangefeed(c *gin.Context) {
 		Type: model.AdminStop,
 	}
 
-	_ = h.capture.OperateOwnerUnderLock(func(owner *owner.Owner) error {
-		owner.EnqueueJob(job)
+	// Use buffered channel to prevernt blocking owner.
+	done := make(chan error, 1)
```
Comment (asddongmen) [inline cdc/api/open.go:323]: May be it is better to extract these codes(line 322 to line 331) into a func like `sendJobToOwner(ctx context)`.

File: cdc/api/open.go [hunk 4]
```diff
@@ -361,11 +366,16 @@ func (h *openAPI) ResumeChangefeed(c *gin.Context) {
 		Type: model.AdminResume,
 	}
 
-	_ = h.capture.OperateOwnerUnderLock(func(owner *owner.Owner) error {
-		owner.EnqueueJob(job)
+	// Use buffered channel to prevernt blocking owner.
```
Comment (asddongmen) [inline cdc/api/open.go:369]: ditto

File: cdc/capture/capture.go [hunk 5]
```diff
@@ -107,13 +107,11 @@ func NewCapture(pdClient pd.Client, kvStorage tidbkv.Storage, etcdClient *etcd.C
...
 
-func NewCapture4Test(isOwner bool) *Capture {
+func NewCapture4Test(o owner.Owner) *Capture {
 	res := &Capture{
 		info: &model.CaptureInfo{ID: "capture-for-test", AdvertiseAddr: "127.0.0.1", Version: "test"},
 	}
-	if isOwner {
-		res.owner = &owner.Owner{}
-	}
+	res.owner = o
```
Comment (hi-rustin) [inline cdc/capture/capture.go:114]: What is the effect if nil is passed in here?

**Review by hi-rustin — COMMENTED** [review]
Will this change break other systems that have cdc cli integrated?(I'm not sure if there are any such customers) Go from asynchronous to synchronous.

File: cdc/owner/owner.go [hunk 6]
```diff
@@ -51,30 +50,45 @@ const (
...
-	done chan struct{}
+	done chan<- error
 }
 
-// Owner manages many changefeeds
-// All public functions are THREAD-SAFE, except for Tick, Tick is only used for etcd worker
-type Owner struct {
+// Owner managers TiCDC cluster.
+//
+// The interface is thread-safe, except for Tick, it's only used by etcd worker.
```
Comment (asddongmen) [inline cdc/owner/owner.go:77]: There is no Tick in this interface.

File: cdc/owner/owner.go [hunk 7]
```diff
@@ -142,7 +157,7 @@ func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState)
 	// when there are different versions of cdc nodes in the cluster,
 	// the admin job may not be processed all the time. And http api relies on
 	// admin job, which will cause all http api unavailable.
-	o.handleJobs()
+	o.HandleJobs()
```
Comment (asddongmen) [inline cdc/owner/owner.go:160]: `handleJobs` in the comment should be change to `HandleJobs`

**Review by asddongmen — APPROVED** [review]

**Comment by overvenus** [comment]
> Will this change break other systems that have cdc cli integrated?(I'm not sure if there are any such customers) Go from asynchronous to synchronous.

It is possible, though I think the impact is neglectable, as owner APIs are fast.

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**